tests/std/tests/P3107R5_enabled_specializations/test.compile.pass.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include <chrono>
+#include <format>
+#include <list>
+#include <ranges>
+#include <stacktrace>
+#include <string>
+#include <string_view>
+#include <thread>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+using namespace std;
+
+struct unoptimized {};
+
+template <class CharT>
+struct std::formatter<unoptimized, CharT> : formatter<string_view, CharT> {
+    template <class FormatContext>
+    auto format(const unoptimized&, FormatContext& ctx) {
+        return formatter<string_view, CharT>::format("unoptimized", ctx);
+    }
+};
+
+static_assert(!enable_nonlocking_formatter_optimization<unoptimized>);
+
+template <class T>
+struct myalloc {
+    using value_type = T;
+
+    template <class U>
+    myalloc(const myalloc<U>&);
+
+    T* allocate(size_t);
+    void deallocate(T*, size_t) noexcept;
+
+    template <class U>
+    bool operator==(const myalloc<U>&) const;
+};
+
+// <chrono> ====================================================================
+static_assert(enable_nonlocking_formatter_optimization<chrono::seconds>);
+static_assert(!enable_nonlocking_formatter_optimization<chrono::duration<unoptimized>>);
+
+static_assert(enable_nonlocking_formatter_optimization<chrono::sys_time<chrono::seconds>>);

```

Comment:
If we need another iteration (if not, I'll add it in a followup):
```suggestion
// Other than `duration` itself, other <chrono> types that depend on `duration` should be unconditionally
// enabled since they decompose `duration`s rather than formatting them directly.
static_assert(enable_nonlocking_formatter_optimization<chrono::sys_time<chrono::seconds>>);
```

Reply:
Added exactly, with you attributed as author. Thanks! :heart_eyes_cat: :scroll: